pkg/auth/serverinterceptors.go

```diff
@@ -48,8 +50,18 @@ func ServerStreamInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.S
 	if UpdateMetrics != nil {
 		UpdateMetrics(ctx)
 	}
-	if err := checkAuth(ctx, info.FullMethod, srv); err != nil {
-		return err
+	if IsTempered {
+		return status.Errorf(
+			codes.DataLoss, "the database should be checked manually as we detected possible tampering")
+	}
+	if !AuthEnabled {
```

Comment:
Why couldn't `&&` used here instad of 3 `if`s?
Also, on master `immuadmin` is only allowed to call from a remote machine if auth is enabled.
This is the check on master: https://github.com/codenotary/immudb/blob/master/pkg/auth/auth.go#L61
```go
if !AuthEnabled {
		isAdminMethod := methodsPermissions[method] == PermissionAdmin
		isLocal := isLocalClient(ctx)
		if !isAdminMethod && (DevMode || isLocal) {
			return nil
		} else if !isLocal {
			return status.Errorf(
				codes.PermissionDenied,
				"server has authentication disabled: only local connections are accepted")
		}
	}
```
**Is `immuadmin` still forbidden to call from a remote machine if auth is off?**

Reply:
Will be reverting to the way it was after the talk with @vchain-us-mgmt